src/app/feature/campaign/campaign.service.ts

```diff
@@ -97,21 +98,33 @@ export class CampaignService {
   }
 
   /**
-   *
-   * @param row
-   * TODO - find better way to link with template actions
-   * TODO - find way to identify any named action list (not just click_action_list)
+   * Utilise template services to process campaign actions
    */
-  public triggerRowActions(row: FlowTypes.Campaign_listRow) {
+  public async triggerRowActions(row: FlowTypes.Campaign_listRow) {
     if (row.click_action_list) {
+      // make a dynamic call to TemplateVariablesService as it also has handling
+      // for @campaigns which would otherwise result in cyclic dependency in constructor
+      const templateVariablesService = this.injector.get(TemplateVariablesService);
+
+      // Process any dynamic variables that might be present in args
+      const parsedActions = [];
       for (const action of row.click_action_list) {
```

Comment:
previous implementation hardcoded only set_field action. New implementation evaluates text for any dynamic conditions before executing within the template-action service